articles/2019-07-09-responsive-sans-media-queries.md

```diff
@@ -0,0 +1,294 @@
+---
+date: 2019-08-27
+title: Du responsive sans media queries
+author: MoOx
+slug: responsive-sans-media-queries
+---
+
+Aujourd'hui, il est assez difficile d'imaginer faire des designs web responsives
+sans avoir recours aux media queries. Cette idée vieille de 1994, devenu
+recommendation du W3C en 2012 (une fois supporté par tous les navigateurs) à
+pris son temps et à sû s'imposer comme l'outil de référence pour faire du design
+adaptatif.
+
+À tel point qu'il parait impossible de faire du responsive sans media queries
+dans l'imaginaire collectif.
+
+## Pourquoi voudrais-tu faire du responsive sans media queries ?
+
+Il faut pas se le cacher: travailler avec les media queries n'est pas toujours
+évident. Cela implique pour chaque "morceau" de votre site ou appli qui va
+devoir s'adapter de prévoir un ou plusieurs breakpoints lié à la taille
+disponible de votre viewport. Ecrire du code lié au viewport pour un "composant"
+bas niveau peut paraître clairement étrange.
+
+Ce côté contre intuitif des MQs m'a toujours dérangé: on se retrouve à cibler
+une taille d’écran, et non pas de cibler la taille disponible pour un élément
+donné.
+
+Lorsque l'on creuse un peu, on tombe souvent sur le concept de "element
+queries". Le rêve de tout intégrateur web. La solution a tous les problèmes
+posés par les media queries.
+
+Franchement écrire du code qui permet à un même composant de se retrouver sur
+une même page a 2 endroits mais avec des dimensions différentes ça serait pas
+cool? Pas qu'un peu.
+
+Alors il y a bien quelques techniques à ce jour notamment
+["les fab four"](https://emails.hteumeuleu.fr/2016/02/fab-four-emails-responsive-sans-media-queries/)
+ou encore des tricks à base de floats ou d'autres trucs plus exotiques, mais
+malheureusement c'est n'est pas toujours maintenable ou intuitif.
+
+Dans notre monde "moderne" (j'en vois déjà certain cracher sur leur écran),
+pourquoi ne pas utiliser JavaScript? (Voilà vous pouvez essuyer votre écran).
+Sérieusement, on pourrait se dire que dans notre contexte, il pourrait être
+pertinent d’utiliser quelque chose comme `window.matchMedia`.
```

Comment:
don't see how that would be a solution

Reply:
What about: "il pourrait être pertinent de simuler des elements queries à coup de `getBoundingClientRect` et d'observer`?

Reply:
yes, ça semble plus accurate déjà